### ISSUE TITLE
KupiaSec - The `lossFee` is always 0 in the `GasProcess.processExecutionFee()` function

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-05-elfi-protocol-judging#211]
KupiaSec

Medium

# The `lossFee` is always 0 in the `GasProcess.processExecutionFee()` function

## Summary

In the `processExecutionFee()` function, the `lossFee` is consistently calculated as 0.

## Vulnerability Detail

Since the `executionFee` was set to the same value as `cache.userExecutionFee`, the `lossFee` is consistently calculated as 0. Consequently, the `totalLossExecutionFee` property of the `CommonData` remains unchanged despite an actual loss occurring.

```solidity
            executionFee = cache.userExecutionFee;
            lossFee = executionFee - cache.userExecutionFee;
```

## Impact

The `totalLossExecutionFee` property of the `CommonData` never changes despite actual losses occurring.

## Code Snippet

https://github.com/sherlock-audit/2024-05-elfi-protocol/blob/main/elfi-perp-contracts/contracts/process/GasProcess.sol#L17-L41

## Tool used

Manual Review

## Recommendation

The calculation of the `lossFee` should be done prior to replacing the `executionFee`.

```diff
-           executionFee = cache.userExecutionFee;
            lossFee = executionFee - cache.userExecutionFee;
+           executionFee = cache.userExecutionFee;
```

Duplicate of #108
